### PR TITLE
Added Consumer Tests for CloudEvents

### PR DESCRIPTION
**Pull request by kerri-lee**
- Added tests for `DeserializeCloudEvents` (deserializing events with system event/custom event payloads, corner cases)
- Addressed the case where JSON-encoded events are not wrapped in an array. For example, when consuming messages received by a Service Bus Queue, events are not wrapped in an array, which would have previously caused an error if passed directly to a Deserialize method.

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridConsumer.cs [hunk 1]
```diff
@@ -75,9 +75,16 @@ private async Task<EventGridEvent[]> DeserializeEventGridEventsInternal(string r
 
             // Deserialize raw JSON string into separate events, deserialize event envelope properties
             JsonDocument requestDocument = await ParseJsonToDocument(requestContent, async, cancellationToken).ConfigureAwait(false);
-            foreach (JsonElement property in requestDocument.RootElement.EnumerateArray())
+            if (requestDocument.RootElement.ValueKind == JsonValueKind.Object)
             {
-                egInternalEvents.Add(EventGridEventInternal.DeserializeEventGridEventInternal(property));
+                egInternalEvents.Add(EventGridEventInternal.DeserializeEventGridEventInternal(requestDocument.RootElement));
+            }
+            else // events are wrapped in an array
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridConsumer.cs:82]: Should we explicitly check for ValueKind.Array?

File: sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridConsumer.cs [hunk 2]
```diff
@@ -172,7 +172,7 @@ private async Task<CloudEvent[]> DeserializeCloudEventsInternal(string requestCo
             {
                 cloudEventsInternal.Add(CloudEventInternal.DeserializeCloudEventInternal(requestDocument.RootElement));
             }
-            else // events are wrapped in an array
+            else if (requestDocument.RootElement.ValueKind == JsonValueKind.Array)
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridConsumer.cs:175]: could the ValueKind ever be anything other than Array or Object?

Comment (kerri-lee) [reply sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridConsumer.cs:175]: In terms of an event that we could deserialize, I don't think so. Should we throw an exception if something else is passed in?

Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridConsumer.cs:175]: I think what you have now is fine. It would just be a no-op if it were some other valueKind. It would be great if we can test for this, though I'm not sure if it is possible.
